src/registrar/models/public_contact.py

```diff
@@ -14,37 +27,126 @@ class ContactTypeChoices(models.TextChoices):
-    # contact's phone number
-    # Must be in ITU.E164.2005 format
-    voice = models.TextField(null=False)
-    # contact's fax number (null ok)
-    # Must be in ITU.E164.2005 format
-    fax = models.TextField(null=True)
-    # contact's authorization code
-    # 16 characters minium
-    pw = models.TextField(null=False)
+    def save(self, *args, **kwargs):
+        """Save to the registry and also locally in the registrar database."""
+        if hasattr(self, "domain"):
+            match self.contact_type:
+                case PublicContact.ContactTypeChoices.REGISTRANT:
+                    self.domain.registrant = self
+                case PublicContact.ContactTypeChoices.ADMINISTRATIVE:
+                    self.domain.administrative_contact = self
+                case PublicContact.ContactTypeChoices.TECHNICAL:
+                    self.domain.technical_contact = self
+                case PublicContact.ContactTypeChoices.SECURITY:
+                    self.domain.security_contact = self
+        super().save(*args, **kwargs)
+
+    contact_type = models.CharField(
+        max_length=14,
+        choices=ContactTypeChoices.choices,
+        help_text="For which type of WHOIS contact",
+    )
+    registry_id = models.CharField(
+        max_length=16,
+        default=get_id,
+        null=False,
+        help_text="Auto generated ID to track this contact in the registry",
+    )
+    domain = models.ForeignKey(
+        "registrar.Domain",
+        on_delete=models.PROTECT,
+        related_name="contacts",
+    )
+
+    name = models.TextField(null=False, help_text="Contact's full name")
+    org = models.TextField(null=True, help_text="Contact's organization (null ok)")
+    street1 = models.TextField(null=False, help_text="Contact's street")
+    street2 = models.TextField(null=True, help_text="Contact's street (null ok)")
+    street3 = models.TextField(null=True, help_text="Contact's street (null ok)")
+    city = models.TextField(null=False, help_text="Contact's city")
+    sp = models.TextField(null=False, help_text="Contact's state or province")
+    pc = models.TextField(null=False, help_text="Contact's postal code")
+    cc = models.TextField(null=False, help_text="Contact's country code")
+    email = models.TextField(null=False, help_text="Contact's email address")
+    voice = models.TextField(
+        null=False, help_text="Contact's phone number. Must be in ITU.E164.2005 format"
+    )
+    fax = models.TextField(
+        null=True,
+        help_text="Contact's fax number (null ok). Must be in ITU.E164.2005 format.",
+    )
+    pw = models.TextField(
+        null=False, help_text="Contact's authorization code. 16 characters minimum."
+    )
+
+    @classmethod
+    def get_default_registrant(cls):
+        return cls(
+            contact_type=PublicContact.ContactTypeChoices.REGISTRANT,
+            registry_id=get_id(),
+            name="CSD/CB – Attn: Cameron Dixon",
```

Comment:
The ADR is https://github.com/cisagov/getgov/blob/main/docs/architecture/decisions/0020-user-models-revisited-plus-whois.md, although it doesn't mention where the info will be stored.

I contemplated `settings.py` but to me that file felt more like *application* settings. I also contemplated a DefaultContact table or whatnot, so that it could be updated in the database without requiring a code deploy, but I suspect this information will not change frequently.

I'm very open to suggestions here and I'll add blurb in our operations doc about how to change the default contacts.